packages/dds/tree/src/indexCommonApi.ts

```diff
@@ -0,0 +1,226 @@
+	type RevertibleAlpha,
+} from "./core/index.js";
+export { type Brand } from "./util/index.js";
+
+export {
+	TreeStatus,
+	TreeCompressionStrategy,
+} from "./feature-libraries/index.js";
+
+export {
+	type ISharedTree,
+	type SharedTreeOptions,
+	ForestType,
+	type SharedTreeContentSnapshot,
+	type SharedTreeFormatOptions,
+	SharedTreeFormatVersion,
+	Tree,
+	type TransactionConstraint,
+	type NodeInDocumentConstraint,
+	type RunTransaction,
+	rollback,
+	type ForestOptions,
+	getBranch,
+	type BranchableTree,
+	type TreeBranchFork,
+	independentInitializedView,
+	type ViewContent,
+	TreeAlpha,
+	independentView,
+} from "./shared-tree/index.js";
+
+export {
+	TreeArrayNode,
+	type Unhydrated,
+	IterableTreeArrayContent,
+	TreeNode,
+	type ViewableTree,
+	type ITree,
+	type TreeNodeSchema,
+	TreeViewConfiguration,
+	type ITreeViewConfiguration,
+	type ITreeConfigurationOptions,
+	type TreeView,
+	type TreeViewEvents,
+	SchemaFactory,
+	type ImplicitFieldSchema,
+	type TreeFieldFromImplicitField,
+	type TreeChangeEvents,
+	type NodeFromSchema,
+	type TreeMapNode,
+	type InsertableTreeNodeFromImplicitAllowedTypes,
+	type TreeLeafValue,
+	FieldKind,
+	FieldSchema,
+	type FieldSchemaMetadata,
+	type ImplicitAllowedTypes,
+	type InsertableTreeFieldFromImplicitField,
+	type InsertableTypedNode,
+	NodeKind,
+	type TreeObjectNode,
+	type TreeNodeFromImplicitAllowedTypes,
+	type TreeNodeSchemaClass,
+	type SchemaCompatibilityStatus,
+	type FieldProps,
+	type InternalTreeNode,
+	type WithType,
+	type NodeChangedData,
+	// Types not really intended for public use, but used in links.
+	// Can not be moved to internalTypes since doing so causes app code to throw errors like:
+	// Error: src/simple-tree/objectNode.ts:72:1 - (ae-unresolved-link) The @link reference could not be resolved: The package "@fluidframework/tree" does not have an export "TreeNodeApi"
+	type TreeNodeApi,
+	type TreeNodeSchemaCore,
+	// Types not really intended for public use, but used in inferred types exposed in the public API.
+	// Can not be moved to internalTypes since doing so causes app code to throw errors like:
+	// error TS2742: The inferred type of 'Inventory' cannot be named without a reference to '../node_modules/@fluidframework/tree/lib/internalTypes.js'. This is likely not portable. A type annotation is necessary.
+	type AllowedTypes,
+	type TreeObjectNodeUnsafe,
+	type InsertableTreeNodeFromImplicitAllowedTypesUnsafe,
+	type TreeArrayNodeUnsafe,
+	type TreeMapNodeUnsafe,
+	type InsertableObjectFromSchemaRecordUnsafe,
+	type InsertableTreeFieldFromImplicitFieldUnsafe,
+	type FieldSchemaUnsafe,
+	type TreeNodeSchemaClassUnsafe,
+	type InsertableTreeNodeFromAllowedTypesUnsafe,
+	// System types (not in Internal types for various reasons, like doc links or cannot be named errors).
+	type typeSchemaSymbol,
+	type TreeNodeSchemaNonClass,
+	// Recursive Schema APIs
+	type ValidateRecursiveSchema,
+	type FixRecursiveArraySchema,
+	// experimental @alpha APIs:
```

Comment:
Here appear to be some of the missing exports:
```suggestion
	type FixRecursiveArraySchema,
	// Index APIs
	type SimpleTreeIndex,
	type IdentifierIndex,
	createSimpleTreeIndex,
	createIdentifierIndex,
	// experimental @alpha APIs:
```